man/systemd.service.xml

```diff
@@ -898,6 +898,26 @@
         </listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><varname>RestartMode=</varname></term>
+
+        <listitem>
+          <para>Takes a string value that specifies how the service should restart:
+            <itemizedlist>
+              <listitem><para>If set to <option>normal</option> (the default), the service restarts by
+              going through a failed state.</para></listitem>
```

Comment:
A service can also enter inactive state rather than failed if `Restart=always` is used.

Reply:
I did a test with the following service file:
```
[Service]
Type=simple
ExecStart=false
Restart=always
```
The service does enter a failed state (`failed-before-auto-restart` to be more precise):
```
May 21 19:47:56 final systemd[1]: test-always.service: Changed dead -> running
May 21 19:47:56 final systemd[1]: test-always.service: Changed running -> failed-before-auto-restart
May 21 19:47:56 final systemd[1]: test-always.service: Changed failed-before-auto-restart -> auto-restart
May 21 19:47:56 final systemd[1]: test-always.service: Changed auto-restart -> dead
```

Reply:
That's because the service is actually failing (since you're calling `false`). You can probably refer to logs in https://github.com/systemd/systemd/issues/27722 for more information (or just change it to `true`).

Reply:
Yeah if you change this to execute `true` instead of `false` it will go through dead-before-auto-restart instead of failed-before-auto-restart which is inactive instead of failed